packages/react-material-ui/src/components/SelectField/SelectField.tsx

```diff
@@ -0,0 +1,78 @@
+'use client';
+
+import React from 'react';
+import {
+  Box,
+  MenuItem,
+  Select,
+  SelectChangeEvent,
+  SelectProps,
+} from '@mui/material';
+
+export const allOption: SelectOption = {
+  value: 'all',
+  label: 'All',
+};
+
+export type SelectOption = {
+  value: string;
+  label: string;
+};
+
+const getStatusValue = (value: string) => {
+  return value === allOption.value ? null : value;
+};
+
+export type SelectFieldProps = {
+  options: SelectOption[];
+  defaultValue: string;
+  label: string;
+  isLoading?: boolean;
+  onChange: (value: string | null) => void;
+} & Omit<SelectProps, 'onChange'>;
+
+const SelectField = ({
+  options = [],
+  defaultValue,
+  label,
+  isLoading = false,
+  onChange,
+  ...rest
+}: SelectFieldProps) => {
+  const handleRenderValue = (value: string) => {
+    if (value === allOption.value) return `${label}: ${allOption.label}`;
+    const optionLabel = optionsWithAll?.find(
+      (label) => label.value === value,
+    )?.label;
+
+    return `${label}: ${optionLabel}`;
+  };
+
+  const handleChange = (event: SelectChangeEvent<string>) => {
+    const value = event.target.value as string;
+    onChange(getStatusValue(value));
+  };
+
+  if (isLoading) {
+    return <Box width={190}>{/* <FormFieldSkeleton hideLabel /> */}</Box>;
```

Comment:
???

Reply:
FormFieldSkeleton depends on #36 
I've merged and fixed this